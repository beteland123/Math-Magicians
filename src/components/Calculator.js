import { useState } from 'react';
import Button from './Button';
import Body from './Body';
import calculate from '../logic/calculate';

function Calculator() {
  // useState hook  to add state to functional components. It takes an initial value as an argument
  const [count, setCount] = useState({ total: null, next: null, operation: null });
  const handleClick = (name) => {
    // returns an array two elements: the current state value and a function to update the state
    setCount(calculate(count, name));
  };
  return (
    <div className="container">
      <Button name={count.total || 0 || count.next} className="result" />
      <Body onClick={() => handleClick()} count={count} setCount={setCount} />
      <Button onClick={() => handleClick('รท')} name="รท" className="opration" />
      <Button onClick={() => handleClick('x')} name="x" className="opration" />
      <Button onClick={() => handleClick('-')} name="-" className="opration" />
      <Button onClick={() => handleClick('+')} name="+" className="opration" />
      <Button onClick={() => handleClick('=')} name="=" className="opration" />
      <Button onClick={() => handleClick('.')} name="." className="opration" />
      <Button onClick={() => handleClick('0')} name="0" className="number" />

    </div>

  );
}
export default Calculator;
