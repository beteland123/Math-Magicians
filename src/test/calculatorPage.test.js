import { render, screen } from '@testing-library/react';
import {
  MemoryRouter, Routes, Route,
} from 'react-router-dom';
import Calculator from '../components/Calculator';

describe('Check Calculator page ', () => {
  test('renders the calculator component with the data when navigating to /calculator', () => {
    render(
      <MemoryRouter initialEntries={['/Calculator']}>
        <Routes>
          <Route path="/Calculator" element={<Calculator />} />
        </Routes>
      </MemoryRouter>,
    );

    expect(screen.getByText('Lets do some math')).toBeInTheDocument();
  });
  test(' it should match the snapshoot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/Calculator']}>
        <Routes>
          <Route path="/Calculator" element={<Calculator />} />
        </Routes>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
