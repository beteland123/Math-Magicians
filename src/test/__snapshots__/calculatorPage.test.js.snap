// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Calculator page   it should match the snapshoot 1`] = `
<div>
  <div
    class="calcPage"
  >
    <h2>
      Lets do some math
    </h2>
    <div
      class="container"
    >
      <input
        class="result"
        readonly=""
        type="text"
        value="0"
      />
      <div
        class="body"
      >
        <button
          class="reset"
          type="button"
        >
          AC
        </button>
        <button
          class="opration"
          type="button"
        >
          +/-
        </button>
        <button
          class="opration"
          type="button"
        >
          %
        </button>
        <button
          class="number"
          type="button"
        >
          7
        </button>
        <button
          class="number"
          type="button"
        >
          8
        </button>
        <button
          class="number"
          type="button"
        >
          9
        </button>
        <button
          class="number"
          type="button"
        >
          4
        </button>
        <button
          class="number"
          type="button"
        >
          5
        </button>
        <button
          class="number"
          type="button"
        >
          6
        </button>
        <button
          class="number"
          type="button"
        >
          1
        </button>
        <button
          class="number"
          type="button"
        >
          2
        </button>
        <button
          class="number"
          type="button"
        >
          3
        </button>
      </div>
      <button
        class="opration"
        type="button"
      >
        รท
      </button>
      <button
        class="opration"
        type="button"
      >
        x
      </button>
      <button
        class="opration"
        type="button"
      >
        -
      </button>
      <button
        class="opration"
        type="button"
      >
        +
      </button>
      <button
        class="opration"
        type="button"
      >
        =
      </button>
      <button
        class="opration"
        type="button"
      >
        .
      </button>
      <button
        class="number"
        type="button"
      >
        0
      </button>
    </div>
  </div>
</div>
`;
